### DISCUSSION TITLE
Cannot zoom to a bounding box

### DISCUSSION THREAD

**Discussion by upswing1** [Mapsui/Mapsui#2310] [Q&A] [answered]
I have an event in a viewmodel whose sole function is to add layers to a map and zoom to a bounding box.

When the event is triggered, the event is handled in code behind and the map adds the layers properly but the map doesn't zoom to the bounding box.  

I tested a known bounding box and it loads if I place the logic in the code behind constructor.  If I place it in the event handler nothing happens. What could be the problem?

**Comment by charlenni** [comment]
When does the zooming take place? While the map is initializing or later?

**Reply by upswing1** [reply]
Later. I initialize the map in the constructor. The maps displays the world. After a few data manipulations, the logic generates an event to refresh the map with the addition of a couple of layers. That works. But the zooming part does not.

**Reply by upswing1** [reply, marked as answer]
I found the problem.  
instead of
//mapControl.Map.Home = n => n.ZoomToBox(BoundingBox);
use
mapControl.Map.Navigator.ZoomToBox(BoundingBox);

Thanks

**Comment by evgenyvalavin** [comment]
For some reason .ZoomToBox() does nothing on my side. Tried specifying different parameters. Nothing.
Version: 5.0.0-beta.7

**Reply by pauldendulk** [reply]
At startup? Or not at all? Do you use map coordinates (which is correct), not latlon coordinates (https://mapsui.com/v5/projections/)?

**Reply by evgenyvalavin** [reply]
I think there was a threading issue because I was invoking .ZoomToBox() as soon as my app starts.
I tested again with some 2 secs delay before invoking .ZoomToBoz() and it works now.
Sorry for the late response

**Reply by pauldendulk** [reply]
hmm, it could still be a bug. The new implementation makes it easier for users but the delayed execution is tricky and could still contain flaws. We may need to investigate this further.
